### ISSUE TITLE
indicate when a diff editor is focused

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#202876]
@jooyoungseo has said that when switching between editors, it's unclear to him when a diff editor has been entered, though this is clear for sighted users.

We should `alert` something like `diff editor` to fix this.

cc @hediet

**Comment by isidorn** [comment]
I think we announce whenever an editor gets focus. Each editor has a title that is announced.
So I am not sure what is missing here? I would not like that we just add a new alert.
Is the title of the diff editor not descriptive enough? Or is nothing getting announced (which would be a bug)

**Comment by meganrogge** [comment]
Good point, the file name is announced and says (Working Tree). @jooyoungseo, is that sufficient to indicate this?

**Comment by meganrogge** [comment]
Closing for now, but will reopen based on your feedback @jooyoungseo

**Comment by meganrogge** [comment]
Just discussed with JooYoung and the redundant info is escaped/ignored by screen reader users, so we need to alert with this info

**Comment by meganrogge** [comment]
Should we consider a better announcement on focus of the text editor? how does this interact with the existing announcement?

**Comment by meganrogge** [comment]
Venkatesh observed that the editor title isn't being read at all, only the accessibility help hint, need to investigate that.

**Comment by meganrogge** [comment]
Edit: we use `status` already. This is working as expected. When a new file is opened, the title is read. When a user focuses the document, they hear the hint, if any, along with the "edit text, ...content". If that hint is disabled, they hear "edit text,...content"

**Comment by meganrogge** [comment]
@jooyoungseo, which command do you use to switch between editor tabs? Is it `View: Open Next Editor in Group`?

I believe this issue is about when using that command, indicating if the focused editor is a diff editor via an `alert`.

**Comment by alexr00** [comment]
What should I be looking for when verifying this?

**Comment by meganrogge** [comment]
Have some editor groups open, including ones with diff editors. Run the command focus next editor group. When a diff editor group is focused, you should hear "diff editor" announced by a screen reader. Disable that setting, `accessibility.verbosity.diffEditorActive` and ensure you don't hear that. 

You should also hear it when navigating the open editors view.

![Image](https://github.com/microsoft/vscode/assets/29464607/ea34f5ae-5af6-4e8d-907b-bc161817a603)

**Comment by alexr00** [comment]
Using the Open Editors view, I don't see anything read about diff editors when I hit `enter` on a diff editor:
![image](https://github.com/microsoft/vscode/assets/38270282/bb867e06-1955-4969-a2f7-88db19ed09a5)

**Comment by alexr00** [comment]
Actually, I do see it when I toggle accessibility mode on 👍
